nautobot_netbox_importer/diffsync/nautobot.py

```diff
@@ -0,0 +1,362 @@
+"""Nautobot DiffSync Importer."""
+
+from typing import Any
+from typing import Dict
+from typing import Iterable
+from typing import MutableMapping
+from typing import Optional
+from typing import Set
+from typing import Tuple
+from typing import Type
+from uuid import UUID
+
+from diffsync import DiffSyncModel
+from django.contrib.contenttypes.fields import GenericForeignKey
+from django.contrib.contenttypes.models import ContentType
+from django.core.exceptions import ValidationError
+from django.db.models import Max
+from nautobot.core.utils.lookup import get_model_from_name
+from pydantic import Field as PydanticField
+
+from .base import EMPTY_VALUES
+from .base import INTERNAL_TYPE_TO_ANNOTATION
+from .base import REFERENCE_INTERNAL_TYPES
+from .base import BaseAdapter
+from .base import ContentTypeStr
+from .base import FieldName
+from .base import InternalFieldTypeStr
+from .base import NautobotBaseModel
+from .base import NautobotBaseModelType
+from .base import RecordData
+from .base import Uid
+from .base import logger
+
+NautobotFields = MutableMapping[FieldName, InternalFieldTypeStr]
+
+# Helper to get the order of models to import. Not all models needs to be here.
+IMPORT_ORDER: Iterable[ContentTypeStr] = (
```

Comment:
What's the plan for maintaining/extending this as Nautobot adds new data models? Will it need to drop support for non-latest versions of Nautobot?

Reply:
Cleaned up and added better description to `IMPORT_ORDER`. It's possible to have currently unsupported content types here, so we can add future content types with keeping old versions support.

The issue will arise once older Nautobot version will need the different order than a newer one.